app/src/main/java/com/example/moviesapp/activity/MainActivity.kt

```diff
@@ -3,16 +3,63 @@ package com.example.moviesapp.activity
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
+import android.view.Menu
+import android.view.View
 import androidx.appcompat.app.AppCompatActivity
 import com.example.moviesapp.R
+import com.example.moviesapp.adapter.PageAdapter
+import com.example.moviesapp.databinding.ActivityMainBinding
+import com.example.moviesapp.utils.Connection
+import com.example.moviesapp.viewmodel.MainActivityViewModel
+import com.google.android.material.tabs.TabLayoutMediator
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class MainActivity : AppCompatActivity(), KoinComponent {
+
+    private val viewModel: MainActivityViewModel by inject()
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var pageAdapter: PageAdapter
+    private var isConnected = false
 
-class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        isConnected = Connection.isOnline(this)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.viewPager.adapter = PageAdapter(this)
+        pageAdapter = binding.viewPager.adapter as PageAdapter
+        TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
+            tab.text = pageAdapter.getPageTitle(position)
+        }.attach()
+
+        viewModel.getMovies(isConnected)
+        viewModel.moviesState.observe({ lifecycle }, ::updateUI)
+    }
+
+    private fun updateUI(moviesData: MainActivityViewModel.MoviesData) {
+        when (moviesData.mState) {
+            MainActivityViewModel.MoviesState.LOADING -> {
+                showLoader()
+            }
+            MainActivityViewModel.MoviesState.EMPTY_DATABASE -> showEmptyState()
+        }
+    }
+
+    private fun showLoader() {
```

Comment:
I left it there because we are going to complete it when we merge my work with Juli's. Given that he had pushed his pr before i pushed my own, we decided to leave it like this and complete it when we united everything.